pkg/k8sutils/job.go

```diff
@@ -442,3 +458,44 @@ func (k8s *K8sImpl) generateEnvFromSecret(env config.Env, namespace string) ([]v
 
 	return generatedEnv, nil
 }
+
+func generateJobLabelsFromEventData(jobName string, actionID string, jsonEventData interface{}) (map[string]string, error) {
+	eventAsMap, ok := jsonEventData.(map[string]interface{})
+	if !ok {
+		return nil, fmt.Errorf("unable to process jsonEventData")
+	}
+
+	keptnContext, ok := eventAsMap["shkeptncontext"].(string)
+	if !ok {
+		return nil, fmt.Errorf("jsonEventData does not contain the field shkeptncontext")
+	}
+
+	eventID, ok := eventAsMap["id"].(string)
+	if !ok {
+		return nil, fmt.Errorf("jsonEventData does not contain the field id")
+	}
+
+	gitCommitID, ok := eventAsMap["gitcommitid"].(string)
+	if !ok {
+		// For legacy events that have no git commit id we just set it to an empty string
+		gitCommitID = ""
+	}
+
+	splitJobName := strings.Split(jobName, "-")
+	jobIndex, err := strconv.Atoi(splitJobName[len(splitJobName)-1])
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse task index")
+	}
+
+	// Job index in the name starts at 1
+	jobIndex = jobIndex - 1
+
```

Comment:
We are introducing a dependency on a string that has been constructed in another package, so we are introducing a dependency on the format of such string between packages. 
We have the task index used for the last part of the job name, which was an int, then we transform it into a string, then we concatenate it, then we split it and re-parse it here.

If we need to manipulate information when creating k8s job we should have all the related code in one package and the inputs should be passed into the package with their original types so that all tightly coupled code is an implementation detail inside one package instead of being a shared responsibility (and dependency!) across packages